### PR TITLE
Feature/averaged submission

### PR DESCRIPTION
**Pull request by ktagowski**

### CODE REVIEW

File: embeddings/evaluator/submission.py [hunk 1]
```diff
@@ -14,19 +17,40 @@
 from embeddings.evaluator.sequence_labeling_evaluator import SequenceLabelingEvaluator
 from embeddings.evaluator.text_classification_evaluator import TextClassificationEvaluator
 from embeddings.utils.json_dict_persister import CustomJsonEncoder
+from embeddings.utils.utils import standardize_name
 
 
 @dataclass
-class Submission:
+class _BaseSubmission(ABC):
```
Comment (djaniak) [inline embeddings/evaluator/submission.py:24]: how about setting `from_local_disk` and `from_wandb` as abstract methods?

File: examples/create_averaged_submissions_from_local_disk.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from embeddings.evaluator.evaluation_results import Task
+from embeddings.evaluator.submission import AveragedSubmission, Submission
+
+app = typer.Typer()
+
+
+def run(
+    dataset: str = typer.Option(..., help="Dataset"),
+    model: str = typer.Option(..., help="Model"),
+    task: Task = typer.Option(..., help="Task type"),
```
Comment (djaniak) [inline examples/create_averaged_submissions_from_local_disk.py:17]: shouldn't we use `typer.Argument` for required fields?

File: embeddings/evaluator/submission.py (hunk 1 above)
Comment (asawczyn) [reply embeddings/evaluator/submission.py:24]: Arguments for both functions are different so mypy does not allow doing it.

File: examples/create_averaged_submissions_from_local_disk.py (hunk 2 above)
Comment (asawczyn) [reply examples/create_averaged_submissions_from_local_disk.py:17]: it is not obligatory. Arguments are used to pass like `main.py some_value`, while options `main.py --name some_value`. 
According to docs: 
> We said before that by default:

>CLI options are optional
>CLI arguments are required
>Again, that's how they work by default, and that's the convention in many CLI programs and systems.

>But you can change that.

>**In fact, it's very common to have optional CLI arguments, it's way more common than having required CLI options.**
https://typer.tiangolo.com/tutorial/arguments/optional/

File: examples/create_averaged_submissions_from_search.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from embeddings.evaluator.submission import AveragedSubmission
+from embeddings.evaluator.submission_utils import filter_hps_summary_runs, filter_retrains
+
+app = typer.Typer()
+
+
+def main(
+    entity: str = typer.Option(..., help="Wandb entity."),
+    project: str = typer.Option(..., help="Wandb project."),
+    task: Task = typer.Option(...),
```
[the remark this answers is not in the document]
Comment (asawczyn) [reply examples/create_averaged_submissions_from_search.py:19]: as above

File: examples/create_submissions_from_search.py [hunk 4]
```diff
@@ -7,11 +8,12 @@
...
+from embeddings.evaluator.submission_utils import filter_hps_summary_runs, filter_retrains
 
 app = typer.Typer()
 
 
-def run(
+def main(
     entity: str = typer.Option(..., help="Wandb entity."),
     project: str = typer.Option(..., help="Wandb project."),
     task: Task = typer.Option(...),
```
[the remark this answers is not in the document]
Comment (asawczyn) [reply examples/create_submissions_from_search.py:19]: above

File: examples/create_averaged_submissions_from_local_disk.py (hunk 2 above)
Comment (djaniak) [reply examples/create_averaged_submissions_from_local_disk.py:17]: Yes, I am aware of this and I also referred to the documentation. I believe that you have used the required CLI option here in the code and that is why I added my comment.

Comment (djaniak) [reply examples/create_averaged_submissions_from_local_disk.py:17]: From my understanding:
- required CLI option is `typer.Optional(...)`
- optional CLI argument is `typer.Argument("some_str")`

File: examples/create_submissions_from_local_disk.py [hunk 5]
```diff
@@ -26,7 +26,7 @@ def run(
         task=task,
         evaluation_file_path=evaluation_file_path,
         packages_file_path=packages_file_path,
-        wandb_log_dir=wandb_log_dir,
+        wandb_config_file_path=wandb_log_dir,
```
Comment (djaniak) [inline examples/create_submissions_from_local_disk.py:29]: I don't like the naming here as we pass a directory where file path is expected

File: examples/create_averaged_submissions_from_local_disk.py [hunk 6]
```diff
@@ -31,14 +31,14 @@ def run(
...
-    wandb_log_dirs_: List[T_path] = list(wandb_log_dirs)
+    wandb_log_dirs_: List[T_path] = list(wandb_config_file_paths)
 
     submission = AveragedSubmission.from_local_disk(
         submission_name=submission_name,
         task=task,
         evaluation_file_paths=evaluation_file_paths_,
         packages_file_paths=packages_file_paths_,
-        wandb_log_dirs=wandb_log_dirs_,
+        wandb_config_file_paths=wandb_log_dirs_,
```
Comment (djaniak) [inline examples/create_averaged_submissions_from_local_disk.py:41]: I don't like the naming here as we pass a directories where file paths are expected

Comment (ktagowski) [reply examples/create_averaged_submissions_from_local_disk.py:41]: This function handles both directories and files

File: examples/create_submissions_from_local_disk.py (hunk 5 above)
Comment (ktagowski) [reply examples/create_submissions_from_local_disk.py:29]: This function handles both directories and files

Comment (djaniak) [reply examples/create_submissions_from_local_disk.py:29]: so how about changing it to `wandb_config_path` instead?

Comment (ktagowski) [reply examples/create_submissions_from_local_disk.py:29]: ok we can change that

**Review by djaniak — DISMISSED** [review]

**Review by djaniak — APPROVED** [review]

**Pull request merged**